ext/mbstring/libmbfl/filters/mbfilter_utf8.c

```diff
@@ -49,10 +51,89 @@ const unsigned char mblen_table_utf8[] = {
 	4, 4, 4, 4, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1
 };
 
+extern int mbfl_bisec_srch2(int w, const unsigned short tbl[], int n);
+extern int mbfl_filt_conv_sjis_mobile_flush(mbfl_convert_filter *filter);
+
+static int mbfl_filt_conv_utf8_mobile_wchar(int c, mbfl_convert_filter *filter);
+static int mbfl_filt_conv_wchar_utf8_mobile(int c, mbfl_convert_filter *filter);
+
+static int mbfl_filt_conv_utf8_wchar(int c, mbfl_convert_filter *filter);
+static int mbfl_filt_conv_wchar_utf8(int c, mbfl_convert_filter *filter);
+static int mbfl_filt_conv_utf8_wchar_flush(mbfl_convert_filter *filter);
+
 static size_t mb_utf8_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf, size_t bufsize, unsigned int *state);
 static void mb_wchar_to_utf8(uint32_t *in, size_t len, mb_convert_buf *buf, bool end);
 static zend_string* mb_cut_utf8(unsigned char *str, size_t from, size_t len, unsigned char *end);
 
+static size_t mb_utf8_docomo_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf, size_t bufsize, unsigned int *state);
+static void mb_wchar_to_utf8_docomo(uint32_t *in, size_t len, mb_convert_buf *buf, bool end);
+static size_t mb_utf8_kddi_a_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf, size_t bufsize, unsigned int *state);
+static void mb_wchar_to_utf8_kddi_a(uint32_t *in, size_t len, mb_convert_buf *buf, bool end);
+static size_t mb_utf8_kddi_b_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf, size_t bufsize, unsigned int *state);
+static void mb_wchar_to_utf8_kddi_b(uint32_t *in, size_t len, mb_convert_buf *buf, bool end);
+static size_t mb_utf8_sb_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf, size_t bufsize, unsigned int *state);
+static void mb_wchar_to_utf8_sb(uint32_t *in, size_t len, mb_convert_buf *buf, bool end);
+
+static bool mbfilter_conv_map_tbl(int c, int *w, const unsigned short map[][3], int n)
```

Comment:
```suggestion
static bool mbfilter_conv_map_tbl(int c, int *w, int n, const unsigned short map[n][3])
```
Maybe? As it took me a while to understand that the ``n`` parameter is used for.

Maybe also clamp down on the size/type of ``n``?

Or is this an issue because those functions are used as function pointers?

Reply:
I have adjusted the order of parameters as suggested. Thanks very much.

Reply:
Changed type to `unsigned int`.